### PR TITLE
Date format

### PR DESCRIPTION
**Pull request by langeyouxiang**
1.Add set the date format in the settings
2.Add date format: MM/dd

### CODE REVIEW

File: source/utils/formatDate.mc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    function formatDate(time as Time.Moment, format as Time.DateFormat?) as FromatDateResult {
+        var englishDays = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"];
+        var englishMonths = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"];
+        var numeralsMonths = ["1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12"];
+        var date = Gregorian.info(time, format);
+
+        var month = date.month;
+        var day = date.day;
+        var enMonthName = englishMonths[month - 1];
+        var numMonthName = numeralsMonths[month - 1];
```
Comment (shbumba) [inline source/utils/formatDate.mc]: Do we really need to convert a day to a string via an array? I thought a regular `date.month.toString()` would be enough

**Review by shbumba — CHANGES_REQUESTED** [review]

File: source/views/DateView.mc [hunk 2]
```diff
@@ -2,18 +2,22 @@ import Toybox.Lang;
...
+        self.updateDateFormatType();
     }
 
     protected function render(drawContext as Dc) as Void {
         var posX = self.getPosX();
         var posY = self.getPosY();
         var width = self.getWidth();
 
-        var dateObj = formatDateToEng(Time.now());
+        var dateObj = FormatDate.formatDateByType(Time.now(), _dateFormatType);
```
Comment (shbumba) [inline source/views/DateView.mc]: Could you change _dateFormatType to self._dateFormatType for transparency

Comment (langeyouxiang) [reply source/views/DateView.mc]: of course :)

File: source/utils/formatDate.mc (hunk 1 above)
Comment (langeyouxiang) [reply source/utils/formatDate.mc]: FromatDateResult.month.toString() won't return different string results depending on the language?
I feel like the object of FromatDateResult.month is date.month.I am not sure about the details, and I need your advice. If there is no problem, I will delete numMonthName.

**Review by shbumba — APPROVED** [review]

**Pull request merged**